watson_developer_cloud/watson_service.py

```diff
@@ -218,11 +251,17 @@ def __init__(self, vcap_services_name, url, username=None, password=None,
                     self.api_key = self.vcap_service_credentials['apikey']
                 if 'api_key' in self.vcap_service_credentials:
                     self.api_key = self.vcap_service_credentials['api_key']
+                if 'iam_api_key' in self.vcap_service_credentials:
+                    self.iam_api_key = self.vcap_service_credentials['iam_api_key']
```

Comment:
If IAM is being used, this will actually show up in VCAP as just `apikey`.

From German's issue:
```
{
  "apikey": "N5msBe7aLCK05w2HSwbrr7iZyMftHMY9iC8-zMVVk804",
  "iam_apikey_description": "Auto generated apikey during resource-key operation for Instance - crn:v1:staging:public:conversation:us-south:a/a2b41967b8284f612fd4b7fd00297819:49014c11-282a-4a74-87ab-8b3dc0777d9b::",
  "iam_apikey_name": "auto-generated-apikey-933a2344-2baa-491c-8ed8-9a5ff1238ed3",
  "iam_role_crn": "crn:v1:bluemix:public:iam::::serviceRole:Manager",
  "iam_serviceid_crn": "crn:v1:staging:public:iam-identity::a/a2b41967b8284f612fd4b7fd00297819::serviceid:ServiceId-de9cbe30-b8ad-40df-b518-884fef8ec713",
  "url": "https://gateway-s.watsonplatform.net/assistant/api"
}
```

so that should be accounted for.